src/backend/distributed/planner/merge_planner.c

```diff
@@ -949,16 +981,32 @@ ConvertSourceRTEIntoSubquery(Query *mergeQuery, RangeTblEntry *sourceRte,
 }
 
 
+/*
+ * ErrorIfMergeHasReturningList raises an exception if the MERGE
+ * has a RETURNING clause.
+ */
+static void
+ErrorIfMergeHasReturningList(Query *query)
+{
+	if (query->returningList)
+	{
+		ereport(ERROR, (errmsg("MERGE with RETURNING is not yet supported")));
+	}
```

Comment:
I'm puzzled by the Codecov warning here, and for the ErrorIfMergeInCopy() function above. The regress test goldfiles clearly show these errors, so why are these warnings being raised do you know ?

Reply:
You are correct, we have exact line coverage. Let's discuss with the infra engineer to see if we are missing anything for PG 17 builds?